bin/reth/src/node/mod.rs

```diff
@@ -271,6 +283,32 @@ impl Command {
     }
 }
 
+/// Loads peers from `file_path` and adds them to the network
+async fn load_peers(file_path: &impl AsRef<Path>, network: &NetworkHandle) -> eyre::Result<()> {
+    let reader = match std::fs::File::open(file_path) {
+        Ok(file) => std::io::BufReader::new(file),
+        Err(e) if e.kind() == ErrorKind::NotFound => return Ok(()),
+        Err(e) => Err(e)?,
+    };
+    info!(target: "reth::cli", file = %file_path.as_ref().display(), "Loading saved peers");
+    let known_peers: Vec<NodeRecord> = serde_json::from_reader(reader)?;
+    known_peers.into_iter().for_each(|record| network.add_peer(record.id, record.udp_addr()));
+    Ok(())
+}
+
+/// Dumps peers to `file_path` for persistence.
+async fn dump_peers(file_path: &impl AsRef<Path>, network: &NetworkHandle) -> eyre::Result<()> {
+    info!(target: "reth::cli", file = %file_path.as_ref().display(), "Saving current peers");
+    let writer = std::io::BufWriter::new(std::fs::File::create(file_path)?);
+    let known_peers = network.peers_handle().all_peers().await;
+
+    let threshold = known_peers.iter().map(|(rep, _)| *rep).max().unwrap_or(0) - 10000;
+    let top_peers: Vec<NodeRecord> =
+        known_peers.into_iter().filter(|(rep, _)| *rep > threshold).map(|(_, node)| node).collect();
+    serde_json::to_writer_pretty(writer, &top_peers)?;
+    Ok(())
+}
```

Comment:
Nit: could move this (along with the if check) to the Drop impl of `NetworkHandle`, this would make sure that it's always used